master/docs/manual/cfg-global.rst

```diff
@@ -597,56 +603,177 @@ This is the main class for statistics service. It is initialized in the master c
+     This callback function should take in two Python datetime objects - a ``start_time`` and an ``end_time`` and return a string that will be sent for storage in the storage backends.
+
+
+
+.. py:class:: buildbot.statistics.capture.CaptureData
+   :noindex:
+
+   Instance of this capture class is for capturing arbitrary data that is not stored as build-data.
+   Needs to be used in conjunction with ``yieldMetricsValue`` (See :ref:`yieldMetricsValue`).
+   Takes the following arguments:
+
+   ``data_name``
+     The name of data to be captured.
+     Same as in ``yieldMetricsValue``.
+   ``builder_name``
+     The name of builder whose times are to be recorded.
+   ``callback=None``
+     The callback function for this class.
+     This callback receives the data sent to  ``yieldMetricsValue`` as ``post_data`` (See :ref:`yieldMetricsValue`).
+     It must return a string that is to be sent to the storage backends for storage.
+
+
+.. py:class:: buildbot.statistics.capture.CaptureDataAllBuilders
+   :noindex:
+
+   Instance of this capture class for capturing arbitrary data that is not stored as build-data on all builders
+   Needs to be used in conjunction with ``yieldMetricsValue`` (See :ref:`yieldMetricsValue`).
+   Takes the following arguments:
+
+   ``data_name``
+     The name of data to be captured.
+     Same as in ``yieldMetricsValue``.
+   ``callback=None``
+     The callback function for this class.
+     This callback receives the data sent to  ``yieldMetricsValue`` as ``post_data`` (See :ref:`yieldMetricsValue`).
+     It must return a string that is to be sent to the storage backends for storage.
+
 
 .. _yieldMetricsValue:
 
 Using ``StatsService.yieldMetricsValue``
 ++++++++++++++++++++++++++++++++++++++++
 
-Advanced users can modify ``BuildSteps`` to use ``StatsService.yieldMetricsValue`` which will send arbitrary data for storage to the ``StatsService``. It takes the following arguments:
+Advanced users can modify ``BuildSteps`` to use ``StatsService.yieldMetricsValue`` which will send arbitrary data for storage to the ``StatsService``.
+It takes the following arguments:
 
-   * ``data_name``: The name of the data being sent or storage.
-   * ``post_data``: A dictionary of key value pair that is sent for storage. The keys will act as columns in a database and the value is stored under that column.
-   * ``buildid``: The integer build id of the current build. Obtainable in all ``BuildSteps``.
+   ``data_name``
+     The name of the data being sent or storage.
+   ``post_data``
+     A dictionary of key value pair that is sent for storage.
+     The keys will act as columns in a database and the value is stored under that column.
+   ``buildid``
+     The integer build id of the current build.
+     Obtainable in all ``BuildSteps``.
 
 Along with using ``yieldMetricsValue``, the user will also need to use the ``CaptureData`` capture class.
+As an example, we can add the following to a build step:
+
+::
```

Comment:
you might want to change it to

```
.. code-block:: python
```

we had quite a few issues when Sphinx/pygments was upgraded.